### DISCUSSION TITLE
[SUGGESTION] avfilters/PyAV approach ?

### DISCUSSION THREAD

**Discussion by MarcoRavich** [cdgriffith/FastFlix#577] [Ideas]
I just discovered the @jeertmans' [avfilters repo](https://github.com/jeertmans/avfilters) which, in [Motivation](https://github.com/jeertmans/avfilters/blob/main/README.md#motivation), claims:

> FFmpeg is a powerful tool for media files manipulation. Many Python libraries already wrap the ffmpeg executable to easily perform audio and video processing in Python.
> 
> However, most of those libraries call ffmpeg through subprocesses. While this is fine in many use cases, this has two major drawbacks:
> 
> 1. you must have ffmpeg installed on your computer, and know the executable path;
> 2. and repeatedly calling subprocesses in a hot loop can be quite slow.
> 
> For performances reasons, a solution is to use bindings to the FFmpeg C libraries, e.g., [PyAV](https://github.com/PyAV-Org/PyAV). Moreover, PyAV ships (by default) with pre-built binary wheels linking against FFmpeg C libraries, so you do not need to rely on a possible local installation of ffmpeg.
> 
> Unfortunately, using PyAV can be quite complex, especially as ffmpeg usually does a lot of auto-corrections that [PyAV](https://github.com/PyAV-Org/PyAV) does not automatically perform.
> 
> This library aims at providing a straighforward ffmpeg-like API for some very common use cases.
> 

Since I'm not a dev, I honestly don't know if it can be a productive/usable approach for FF goals, anyway I believe that an exchange of knowledge can still be useful.

**Comment by jeertmans** [comment]
Hello, while what I say in the README is, I believe, very true, I have to mention that the avfilters package is currently very light, as I didn’t have time (nor needed) for more complex utilities. One example is that it may fail to handle (multiple) audio tracks or subtitles. PyAV, on the other hand, is more complex but provides way more features.

I know nothing about your package, but I’d be glad to provide any help or comment :)
